pcp/PCPDynamicMeter.c

```diff
@@ -65,13 +66,14 @@ static void PCPDynamicMeter_parseMetric(PCPDynamicMeters* meters, PCPDynamicMete
       /* use derived metrics in dynamic meters for simplicity */
       char* error;
       if (pmRegisterDerivedMetric(metric->name, value, &error) < 0) {
-         char note[1024];
-         xSnprintf(note, sizeof(note),
+         char* note;
+         xAsprintf(&note,
                    "failed to parse expression in %s at line %u\n%s\n%s",
                    path, line, error, pmGetProgname());
```

Comment:
Customarily the program name is prefixed in an error message. So what about this?
```suggestion
                   "%s: Failed to parse expression in %s at line %u\n%s\n",
                   pmGetProgname(), path, line, error);
```